### DISCUSSION TITLE
loadAnimationTrigger with SynchronouslyBlockingCurrentThread cause animation to disappear

### DISCUSSION THREAD

**Discussion by denandreychuk** [airbnb/lottie-ios#2125] [General]
```
LottieView(
    dotLottieFile: try? .SynchronouslyBlockingCurrentThread.named(someAnimation).get()
)
.loadAnimationTrigger(.constant(isPlaying))
```
After 'isPlaying' changes, it causes the animation to disappear.

**Comment by calda** [comment]
This is the intended behavior of the `loadAnimationTrigger` API. 

The idea is that you should only call the `loadAnimationTrigger` when you expect the closure to provide a different animation, since otherwise there is usually no benefit to reloading the animation. Why are you calling `loadAnimationTrigger` here?

**Reply by calda** [reply]
Ah see, I misread your example and didn't notice that you're passing a `dotLottieFile` explicitly instead of using the `loadAnimation` closure. I think this is a bug then. If you used the load animation closure syntax instead I think this would work as expected:

```
LottieView {
  try await DotLottieFile.named(someAnimation)
}
.loadAnimationTrigger(.constant(isPlaying))
```

**Reply by denandreychuk** [reply]
Yeap, I can confirm that using closure is working correctly. Thanks!
